### ISSUE TITLE
Validating config changes returns an empty error and response

### ISSUE THREAD

**Issue by aodj** [bosun-monitor/bosun#1430]
When attempting to write an alert in the UI, I receive the message `"Error validation:"` with no further information. The query I'm attempting to write is as follows, and is only marginally modified from the auto generated one made by clicking on the `Rule` button in the `Expression` tab (which also generates the same error)

```
# backends
influxHost = influxdb:8086
logstashElasticHosts = http://elasticsearch:9200

# settings
emailFrom = bosun@alerting.com
hostname = bosun
smtpHost = hammer

template redis.mem_free.low {
  subject = {{.Last.Status}}: {{.Alert.Name}} on {{.Group.host}}
  body = `<p>Name: {{.Alert.Name}}
  <p>Tags:
  <table>
    {{range $k, $v := .Group}}
      <tr><td>{{$k}}</td><td>{{$v}}</td></tr>
    {{end}}
  </table>`
}

alert redis.mem_free.low {
  template = redis.mem_free.low
  $query = min(influx("telegraf", "SELECT value FROM mem_free WHERE service = 'redis' GROUP BY host", "5m", ""))
  crit = $query < 1073741824
}
```

The Chrome console reports an empty response but nothing further.

Can anyone provide some insight, because the available docs don't say much about using Influx or ElasticSearch, at this time.

**Comment by captncraig** [comment]
Interesting. When I run that, I get `conf: test:24:2: at <crit = $query < 1073...>: expr: parse: not enough arguments for influx` What version of bosun are you running (`/api/version`)

**Comment by aodj** [comment]
`bosun version 0.4.0 (9ef650b36b5d4abb6cf6fe4a4a57327045a97fa4) built 2015-09-18T14:39:03Z`

**Comment by gbrayut** [comment]
Influx now has a 3rd time argument for grouping. Probably was added after the 0.4.0 release

**Comment by giganteous** [comment]
I can confirm that; "Full" support for querying Influx was only fixed just after the 0.4.0 cut.

So, in 0.4.0 there is a known bug: You can't group.

**Comment by aodj** [comment]
Does that mean that the Influx support in the 0.4.0 release is essentially non functional?

**Comment by gbrayut** [comment]
0.4.0 was the first release with any influxdb support, and the group by support was not included until after 0.4.0 was released. If you build bosun from the current master branch or from one of the last two closed prs at 
https://github.com/bosun-monitor/bosun/pulls?q=label%3Ainfluxdb+is%3Apr+is%3Aclosed it should support group by but still may have some bugs since it is very new and we don't use it as a data store at stack Overflow.

Bosun has recently transitioned from storing internal state in boltdb to using ledis or redis. We probably will have a new pre release build in the next week or so, but it may take a few more weeks to get docs and everything ready for a full release.

**Comment by captncraig** [comment]
Influx should work from master or from the 0.5.0 pre-release: https://github.com/bosun-monitor/bosun/releases/tag/0.5.0-alpha
